src/features/Dashboard/WeeklySchedule/index.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { Schedule } from 'react-paragon-topaz';
+
+import { startOfWeek, endOfWeek, isWithinInterval } from 'date-fns';
+
+import { fetchAllClassesData } from 'features/Common/data';
+
+import { formatDateRange } from 'helpers';
+import 'features/Dashboard/WeeklySchedule/index.scss';
+
+const WeeklySchedule = () => {
+  const dispatch = useDispatch();
+  const username = useSelector((state) => state.main.username);
+  const classesData = useSelector((state) => state.common.allClasses.data);
+  const [classList, setClassList] = useState([]);
+  const startCurrentWeek = startOfWeek(new Date());
+  const endCurrentWeek = endOfWeek(new Date());
+  const [stateDate, setStateDate] = useState([
+    {
+      startDate: startCurrentWeek,
+      endDate: endCurrentWeek,
+      key: 'selection',
+      color: '#e4faff',
+    },
+  ]);
```

Comment:
I think `startCurrentWeek` and `endCurrentWeek` are unnecessary,  you aren't use it in anywhere else, instead why just:

```suggestion
  const [stateDate, setStateDate] = useState([
    {
      startDate: startOfWeek(new Date()),
      endDate: endOfWeek(new Date()),
      key: 'selection',
      color: '#e4faff',
    },
  ]);
```